### PR TITLE
boards/b-l072z-lrwan1: add ADC definitions

### PR DESCRIPTION
**Pull request by ichatz**
### Contribution description

Updated the boards/b-l072z-lrwan1/ directory to define the ADC wiring.

### Testing procedure

Simply use the adctest with the A0 or A2 ardiuono connectors.

### CODE REVIEW

File: boards/b-l072z-lrwan1/Kconfig [hunk 1]
```diff
@@ -14,6 +14,7 @@ config BOARD_B_L072Z_LRWAN1
     select CPU_MODEL_STM32L072CZ
 
     # Put defined MCU peripherals here (in alphabetical order)
+    select HAS_PERIPH_ADC
```
Comment (benpicco) [inline boards/b-l072z-lrwan1/Kconfig:17]:
```suggestion
    select HAS_PERIPH_ADC
```

CI has an issue with the [trailing whitespace](https://github.com/RIOT-OS/RIOT/pull/15866/checks?check_run_id=1776409828).

You can use `git commit --amend` to add the change to the previous commit, then just `git push -f` to force push it.

File: boards/b-l072z-lrwan1/include/periph_conf.h [hunk 2]
```diff
@@ -147,6 +147,27 @@ static const spi_conf_t spi_config[] = {
...
+ * @{
+ */
+static const adc_conf_t adc_config[] = {
+    {GPIO_PIN(PORT_A, 0), 0},
+    {GPIO_PIN(PORT_A, 1), 1},
+    {GPIO_PIN(PORT_A, 4), 4},
+    {GPIO_PIN(PORT_B, 0), 0},
+    {GPIO_PIN(PORT_C, 1), 1},
+    {GPIO_PIN(PORT_C, 0), 0},
+};
```
Comment (aabadie) [inline boards/b-l072z-lrwan1/include/periph_conf.h:166]: I'm comparing the pin configured with UM2115. Are you sure this is working ? Especially A1, A3-A5 ?

image

Comment (ichatz) [reply boards/b-l072z-lrwan1/include/periph_conf.h:166]: Indeed in this board only A0 and A2 are accessible. Though, I made the change to match the table.

File: boards/b-l072z-lrwan1/include/periph_conf.h [hunk 3]
```diff
@@ -147,6 +147,27 @@ static const spi_conf_t spi_config[] = {
...
+ *
+ * Note that we do not configure all ADC channels,
+ * we just define 6 ADC channels, for the Nucleo
+ * Arduino header pins A0-A5
+ *
+ * @{
+ */
+static const adc_conf_t adc_config[] = {
+    { GPIO_PIN(PORT_A, 0), 0 },
+    { GPIO_PIN(PORT_A, 0), 0 },
```
Comment (benpicco) [inline boards/b-l072z-lrwan1/include/periph_conf.h:161]: Why is this here twice?

File: boards/b-l072z-lrwan1/include/periph_conf.h [hunk 4]
```diff
@@ -147,6 +147,27 @@ static const spi_conf_t spi_config[] = {
...
+    { GPIO_PIN(PORT_A, 4), 4 },
+    { GPIO_PIN(PORT_A, 4), 4 },
```
Comment (benpicco) [inline boards/b-l072z-lrwan1/include/periph_conf.h:163]: same here

**Review by benpicco — COMMENTED** [review]
Please squash the commits and only keep the ADC changes - please drop the whitespace changes to the rest of the file.

**Comment by fjmolinas** [comment]
ping @ichatz
